frontend/app/src/components/wallet-bridge/Pairing.vue

```diff
@@ -0,0 +1,345 @@
+import { useSupportedChains } from '@/composables/info/chains';
+import { useWalletHelper } from '@/composables/trade/wallet-helper';
+import { EIP155, ROTKI_DAPP_METADATA, useWalletStore } from '@/store/trade/wallet';
+import { type IWalletKit, WalletKit, type WalletKitTypes } from '@reown/walletkit';
+import { assert } from '@rotki/common';
+import { get, set } from '@vueuse/core';
+import { Core } from '@walletconnect/core';
+import { formatJsonRpcError } from '@walletconnect/jsonrpc-utils';
+import { buildApprovedNamespaces, getSdkError } from '@walletconnect/utils';
+import { onMounted, ref } from 'vue';
+
+interface LogEntry {
+  message: string;
+  timestamp: string;
+  type: 'info' | 'success' | 'error';
+}
+
+const props = defineProps<{
+  address: string;
+  connectedChainId: number;
+}>();
+
+const { address, connectedChainId } = toRefs(props);
+
+const { t } = useI18n();
+const projectId = import.meta.env.VITE_WALLET_CONNECT_PROJECT_ID as string;
+
+const core = new Core({
+  projectId,
+});
+
+const pairUri = ref('');
+const logs = ref<LogEntry[]>([]);
+const isConnecting = ref(false);
+const walletKit = ref<IWalletKit | null>(null);
+const activeSessions = ref<SessionTypes.Struct[]>([]);
+
+const { getBrowserProvider } = useWalletStore();
+const { getChainFromChainId, getEip155ChainId } = useWalletHelper();
+const { getChainName } = useSupportedChains();
+
+function addLog(message: string, type: 'info' | 'success' | 'error' = 'info') {
+  const timestamp = new Date().toLocaleTimeString();
+  set(logs, [{ message, timestamp, type }, ...get(logs)]);
+}
+
+async function initializeWalletKit() {
+  try {
+    set(walletKit, await WalletKit.init({
+      core,
+      metadata: ROTKI_DAPP_METADATA,
+    }));
+    setupListeners();
+
+    addLog('WalletKit initialized successfully', 'success');
+  }
+  catch (error) {
+    addLog(`Failed to initialize WalletKit: ${(error as Error).message}`, 'error');
+  }
+}
+
+async function pair() {
+  const pairUriVal = get(pairUri).trim();
+  if (!pairUriVal) {
+    addLog('Please enter a valid Pair URI', 'error');
+    return;
+  }
+
+  set(isConnecting, true);
+
+  try {
+    const kit = get(walletKit);
+    if (!kit) {
+      return;
+    }
+
+    if (!kit) {
+      throw new Error('WalletKit not initialized');
+    }
+
+    // Connect using the URI
+    await kit.core.pairing.pair({ uri: pairUriVal });
+    addLog('Successfully paired with the Electron app, you can do transaction there.', 'success');
+
+    set(pairUri, '');
+  }
+  catch (error) {
+    addLog(`Connection failed: ${(error as Error).message}`, 'error');
+  }
+  finally {
+    set(isConnecting, false);
+  }
+}
+
+function getActiveSessions(): SessionTypes.Struct[] {
+  const kit = get(walletKit);
+  if (!kit)
+    return [];
+
+  const sessionsMap = kit.getActiveSessions() || {};
+  return Object.values(sessionsMap);
+}
+
+async function chainChanged(topic: string, chainId: string) {
+  const kit = get(walletKit);
```

Comment:
wouldn't it make sense to bail early here if the kit is not defined? and only proceed if it is?